internal/agent-api/client.go

```diff
@@ -162,6 +166,19 @@ func (a *AgentClient) Undeploy() error {
 	return nil
 }
 
+func (a *AgentClient) RecordExecTime(elapsedNanos int64) {
+	a.execTotalNanos += elapsedNanos
```

Comment:
This isn't thread-safe. If a function is called at the same exact moment, there will be a race to set this.

Reply:
Could simply add an execTimeMutex in the agent and lock/defer unlock before `a.execTotalNanos += elapsedNanos`

Reply:
if this codebase gets one more mutex, I'm rewriting all the async code ;)

Reply:
Yeah, `atomic` is a better way to do it ;)